hawkbit-repository/hawkbit-repository-jpa/src/main/java/org/eclipse/hawkbit/repository/jpa/rsql/RSQLUtility.java

```diff
@@ -594,19 +610,59 @@ private Predicate getInPredicate(final List<Object> transformedValues, final Pat
             }
         }
 
-        private Predicate getOutPredicate(final List<Object> transformedValues, final Path<Object> fieldPath) {
-            final List<String> outParams = new ArrayList<>();
-            for (final Object param : transformedValues) {
-                if (param instanceof String) {
-                    outParams.add(((String) param).toUpperCase());
-                }
+        private Predicate getOutPredicate(final List<Object> transformedValues, final String finalProperty,
+                final A enumField, final Path<Object> fieldPath) {
+
+            final String[] fieldNames = getSubAttributesFrom(finalProperty);
+            final List<String> outParams = transformedValues.stream().filter(String.class::isInstance)
+                    .map(String.class::cast).map(String::toUpperCase).collect(Collectors.toList());
+
+            if (isSimpleField(fieldNames, enumField.isMap())) {
+                return toNullOrNotInPredicate(fieldPath, transformedValues, outParams);
             }
-            if (!outParams.isEmpty()) {
-                return cb.not(cb.upper(pathOfString(fieldPath)).in(outParams));
+
+            clearOuterJoinsIfNotNeeded();
+
+            return toOutWithSubQueryPredicate(fieldNames, transformedValues, enumField, outParams);
+        }
+
+        private Predicate toNullOrNotInPredicate(final Path<Object> fieldPath, final List<Object> transformedValues,
+                final List<String> outParams) {
+
+            Predicate inPredicate;
+            final Path<String> pathOfString = pathOfString(fieldPath);
+
+            if (outParams.isEmpty()) {
+                inPredicate = fieldPath.in(transformedValues);
```

Comment:
could you use the ternary operator or make the inPredicate final (it is being always assigned)